### PR TITLE
FEATURE: show assigned user on search results

### PR DESCRIPTION
**Pull request by techAPJ**
Related PR on core: https://github.com/discourse/discourse/pull/11428

### CODE REVIEW

File: plugin.rb [hunk 1]
```diff
@@ -346,6 +346,19 @@ class ::ListController
     end
   end
 
+  add_to_class(:search_topic_list_item_serializer, :assigned_to_user_id) do
+    id = object.custom_fields[TopicAssigner::ASSIGNED_TO_ID]
+    id && id.to_i rescue nil
```
Comment (ZogStriP) [inline plugin.rb:351]: I don't think `to_i` ever raise an exception. `nil.to_i == 0`

Comment (techAPJ) [reply plugin.rb:351]: Looking at code comment:

https://github.com/discourse/discourse-assign/blob/c4aefd5917be8f5b897cfede19fd120deb3c8a6c/plugin.rb#L394-L395

I think it's for array (`[].to_i`) cases.

Comment (ZogStriP) [reply plugin.rb:351]: I see, I guess it might be worth adding the same comment?

Comment (techAPJ) [reply plugin.rb:352]: Makes sense, done.

**Review by ZogStriP — APPROVED** [review]

File: plugin.rb (hunk 1 above)
Comment (tgxworld) [reply plugin.rb:352]: I highly recommend never using `rescue nil` because it rescues all and any type of errors.

Comment (eviltrout) [reply plugin.rb:352]: I've seen you recommend this before (and correctly.) I wonder if we can lint for it?

**Pull request merged**